src/tss2-tcti/tcti-mssim.c

```diff
@@ -562,17 +524,41 @@ Tss2_Tcti_Mssim_Init (
         return TSS2_RC_SUCCESS;
     }
 
-    rc = conf_str_to_host_port (uri_str, hostname, &port);
-    if (rc != TSS2_RC_SUCCESS) {
-        return rc;
+    if (conf != NULL) {
+        LOG_TRACE ("conf is not NULL");
+        if (strlen (conf) > TCTI_MSSIM_CONF_MAX) {
+            LOG_WARNING ("Provided conf string exceeds maximum of %u",
+                         TCTI_MSSIM_CONF_MAX);
+            return TSS2_TCTI_RC_BAD_VALUE;
+        }
+        conf_copy = strdup (conf);
+        if (conf_copy == NULL) {
+            LOG_ERROR ("Failed to allocate buffer: %s", strerror (errno));
+            rc = TSS2_TCTI_RC_GENERAL_FAILURE;
+            goto fail_out;
+        }
+        LOG_DEBUG ("Dup'd conf string to: 0x%" PRIxPTR,
+                   (uintptr_t)conf_copy);
+        rc = parse_key_value_string (conf_copy,
+                                     mssim_kv_callback,
+                                     &mssim_conf);
+        if (rc != TSS2_RC_SUCCESS) {
+            goto fail_out;
+        }
     }
+    LOG_DEBUG ("Initializing mssim TCTI with host: %s, port: %" PRIu16,
+               mssim_conf.host, mssim_conf.port);
 
-    rc = socket_connect (hostname, port, &tcti_mssim->tpm_sock);
+    rc = socket_connect (mssim_conf.host,
+                         mssim_conf.port,
+                         &tcti_mssim->tpm_sock);
     if (rc != TSS2_RC_SUCCESS) {
-        return rc;
+        goto fail_out;
     }
 
-    rc = socket_connect (hostname, port + 1, &tcti_mssim->platform_sock);
+    rc = socket_connect (mssim_conf.host,
```

Comment:
We should probably initialize
tcti_mssim->tpm_sock = -1 and 
tcti_mssim->platform_sock = -1
here before calling the first socket_connect.

Reply:
Same response I gave to @danintel yesterday (https://github.com/tpm2-software/tpm2-tss/pull/976#discussion_r183529819) applies: please limit your review to code in the patch. I don't disagree with with the *issue* you're bringing up, just that this isn't the right place to do it. IMHO the right thing to do here is to create an issue to track the work, bonus points if you assign it to yourself and knock out a patch :wink: